### PR TITLE
OCPVE-663: feat: introduce k8s events to lvmcluster and vgmanager

### PR DESCRIPTION
**Pull request by jakobmoellerdev**
This introduces event recorders to the lvm cluster controller as well as vg manager and issues events on the most important events and failures related to these resources.

### CODE REVIEW

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

File: pkg/vgmanager/vgmanager_controller.go [hunk 1]
```diff
@@ -48,6 +51,18 @@ const (
 	metadataWarningPercentage = 95
 )
 
+type EventReasonInfo string
+type EventReasonError string
+
+const EventReasonErrorNoAvailableDevicesForVG EventReasonError = "NoAvailableDevicesForVG"
```
[the remark this answers is not in the document]
Comment (jakobmoellerdev) [reply pkg/vgmanager/vgmanager_controller.go:57]: I think its go convention to write enum strings like that. I can remove it but if you look at other constant enums they should look similar. I agree though they are too long...

**Pull request merged**